cnf-tests/testsuites/e2esuite/dpdk/numa_node_sriov.go

```diff
@@ -0,0 +1,226 @@
+		if isSNO {
+			Skip("Single Node openshift not yet supported")
+		}
+
+		By("Clean any pods in " + sriovnamespaces.Test + " namespace")
+		namespaces.CleanPods(sriovnamespaces.Test, sriovclient)
+	})
+
+	BeforeAll(func() {
+
+		err := namespaces.Create(sriovnamespaces.Test, client.Client)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Clean SRIOV policies and networks")
+		networks.CleanSriov(sriovclient)
+
+		By("Discover SRIOV devices")
+		sriovCapableNodes, err := sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(sriovCapableNodes.Nodes)).To(BeNumerically(">", 0))
+		testingNode, err := nodes.GetByName(sriovCapableNodes.Nodes[0])
+		Expect(err).ToNot(HaveOccurred())
+		By("Using node " + testingNode.Name)
+
+		By("Apply single-numa-node performance profile")
+		perfProfile := performanceprofile.DefineSingleNUMANode("single-numa-node-pp", machineConfigPoolName)
+		err = performanceprofile.OverridePerformanceProfile("single-numa-node-pp", machineConfigPoolName, perfProfile)
+		Expect(err).ToNot(HaveOccurred())
+
+		sriovDevices, err := sriovCapableNodes.FindSriovDevices(testingNode.Name)
+		Expect(err).ToNot(HaveOccurred())
+
+		numa0Device, err := findDeviceOnNUMANode(testingNode, sriovDevices, "0")
+		Expect(err).ToNot(HaveOccurred())
+		By("Using NUMA0 device " + numa0Device.Name)
+
+		numa1Device, err := findDeviceOnNUMANode(testingNode, sriovDevices, "1")
+		Expect(err).ToNot(HaveOccurred())
+		By("Using NUMA1 device " + numa1Device.Name)
+
+		// SriovNetworkNodePolicy
+		// NUMA node0 device excludeTopology = true
+		// NUMA node0 device excludeTopology = false
+		// NUMA node1 device excludeTopology = true
+		// NUMA node1 device excludeTopology = false
+
+		By("Create SRIOV policies and networks")
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa0Device.Name+"#0-3"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-0-exclude-topology-false"),
+			withExcludeTopology(false),
+		)
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa0Device.Name+"#4-7"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-0-exclude-topology-true"),
+			withExcludeTopology(true),
+		)
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa1Device.Name+"#0-3"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-1-exclude-topology-false"),
+			withExcludeTopology(false),
+		)
+
+		createSriovNetworkAndPolicy(
+			withNodeSelector(testingNode),
+			withNumVFs(8), withPfNameSelector(numa1Device.Name+"#4-7"),
+			withNetworkNameAndNamespace(sriovnamespaces.Test, "test-numa-1-exclude-topology-true"),
+			withExcludeTopology(true),
+		)
+
+		By("Waiting for SRIOV devices to get configured")
+		networks.WaitStable(sriovclient)
+	})
+
+	AfterAll(func() {
```

Comment:
In BeforeAll, a namespace is created. Should it not be deleted here?

If CI runs test suites in parallel, the same namespace may already exist in which case either
  a) we don't want to delete it here at the cost of leaving it as a leftover or
  b) we create its own namespace and so we can freely delete it here

Reply:
Good point. The problem here is that `"sriov-conformance-testing"` namespace is a little bit special:

- It must exist to make the DiscoverSriov() function work. (es [s2i.go#L163](https://github.com/openshift-kni/cnf-features-deploy/blob/dad3bccf61b7b33ed1319a69f8810745157af7a4/cnf-tests/testsuites/e2esuite/s2i/s2i.go#L163) ). We need to clean that in the upstream repository.

- It's dumped by the k8sreporter (see [pkg/utils/reporter.go#L97](https://github.com/openshift-kni/cnf-features-deploy/blob/efe8c5ddcad32b5d0b8c58f51d391aafc96fe77d/cnf-tests/testsuites/pkg/utils/reporter.go#L97)), though it wouldn't be a problem to an entry like "numa-dpdk-tests-ns": "sriov"

- it's cleaned by the Fixture mechanism (see [pkg/features/features.go#L103](https://github.com/openshift-kni/cnf-features-deploy/blob/efe8c5ddcad32b5d0b8c58f51d391aafc96fe77d/cnf-tests/testsuites/pkg/features/features.go#L103)), as cleaning it in the AfterAll/AfterEach makes the k8sreporter not to gather it.

None of the above points represents a hard limit for using the own namespace, but they require a reasonable effort.